### ISSUE TITLE
Credits are added to auto-generated excerpts even when list_on_excerpts is false

### ISSUE THREAD

**Issue by markjaquith** [webgilde/image-source-control#102]
I have `list_on_excerpts` false, but noticed the image credits were getting injected on autogenerated excerpts (i.e. no manual excerpt... excerpt generated from post content).

I worked around it like so:

```php

function iscPluginExists() {
	return method_exists(\ISC_Public::class, 'get_instance') && method_exists(\ISC_Public::class, 'content_filter');
}

function removeIscContentFilter() {
	if (iscPluginExists()) {
		remove_filter('the_content', [\ISC_Public::get_instance(), 'content_filter'], 20);
	}
}

function restoreIscContentFilter() {
	if (iscPluginExists()) {
		add_filter('the_content', [\ISC_Public::get_instance(), 'content_filter'], 20);
	}
}

// Do not run ISC when autogenerating excerpts.
// We do this by recognizing when the get_the_excerpt filter is being applied, removing the
// ISC filter on the_content at that point, and then restoring it when get_the_excerpt is done.
add_filter('get_the_excerpt', function($input) {
	removeIscContentFilter();
	return $input;
}, -999);
add_filter('get_the_excerpt', function($input) {
	restoreIscContentFilter();
	return $input;
}, 999);
```

But it seems like `list_on_excerpts` should also prevent the `the_content` filter from generating credits when run from within the `get_the_excerpt` filter.

Probably the best way to do this inside the plugin is to have ISC_Public store a property `isGeneratingExcerpt` or such, that gets turned on at `get_the_excerpt:-999` and turned off at `get_the_excerpt:999`, and then the `content_filter()` method should bail early if `list_on_excerpts` is empty and the `isGeneratingExcerpt` flag is true.

I'm happy to submit a PR if you agree and will accept this change.

**Comment by webzunft** [comment]
Thanks for reporting this @markjaquith.
I tried to reproduce an issue and here is what I came up with:
- using the "Overlay" source option
- the theme uses `the_excerpt()` on post list pages
- an image is displayed at the top of the post

In this case, I saw the image source in the excerpt. Is that the scenario you are trying to resolve?
If so, I would probably remove the `the_content` filter for excerpts completely, regardless the `list_on_excerpts` option. When adding that option, I was just thinking about the source list below the content, not overlays. I seem so just have missed that edge case and without images displaying in the excerpt, there is no need to keep the overlay content.

**Comment by webzunft** [comment]
@markjaquith Please see https://github.com/webgilde/image-source-control/pull/103/files. Compared with your suggested code I have the feeling that I might have missed something.

**Comment by markjaquith** [comment]
>  Is that the scenario you are trying to resolve?

Yes!

> If so, I would probably remove the `the_content` filter for excerpts completely, regardless the `list_on_excerpts` option. 

I agree. I assumed different intent for the `list_on_excerpts` option (I admit I was looking at the options array we'd hardcoded for this site, and not at the UI). I can't think of a reason you'd want image captions in excerpts, so your fix in #103 seems good.
